Tests/AppStoreConnectClientTests/RetryingMiddlewareTests.swift

```diff
@@ -0,0 +1,250 @@
+            return (HTTPResponse(status: status), nil)
+        }
+        let baseURL = try XCTUnwrap(URL(string: "http://example.com"), "Unwrap fail")
+        let (response, _) = try await sut.intercept(
+            HTTPRequest(
+                method: .get,
+                scheme: "http",
+                authority: "example.com",
+                path: "/test"
+            ),
+            body: HTTPBody([1, 2, 3], length: .known(3), iterationBehavior: .single),
+            baseURL: baseURL,
+            operationID: "testOperation",
+            next: next
+        )
+        XCTAssertEqual(response.status.code, internalServerError)
+        await fulfillment(of: [expectation], timeout: 3.0)
+    }
+    
+    func testRepeatWhenIterationBehaviorMultipleAndPolicyRepeat() async throws {
+        let sut = RetryingMiddleware(
+            signals: [.code(internalServerError)],
+            policy: .upToAttempts(count: 3)
+        )
+        let expectation = XCTestExpectation(
+            description: "Expect 'next' to be called multiple times due to 'multiple' iteration behavior and retry policy"
+        )
+        expectation.expectedFulfillmentCount = 3
+        let next: @Sendable (
+            HTTPRequest,
+            HTTPBody?,
+            URL
+        ) async throws -> (HTTPResponse, HTTPBody?) = { _, _, _ in
+            expectation.fulfill()
+            let status = HTTPResponse.Status(code: self.internalServerError)
+            return (HTTPResponse(status: status), nil)
+        }
+        let baseURL = try XCTUnwrap(URL(string: "http://example.com"), "Unwrap fail")
+        let (response, _) = try await sut.intercept(
+            HTTPRequest(
+                method: .get,
+                scheme: "http",
+                authority: "example.com",
+                path: "/test"
+            ),
+            body: HTTPBody([1, 2, 3], length: .known(3), iterationBehavior: .multiple),
+            baseURL: baseURL,
+            operationID: "testOperation",
+            next: next
+        )
+        XCTAssertEqual(response.status.code, internalServerError)
+        await fulfillment(of: [expectation], timeout: 3.0)
+    }
+    
+    func testIfSignalsContainsError() async throws {
+        let sut = RetryingMiddleware(
+            signals: [.errorThrown],
+            policy: .upToAttempts(count: 3)
+        )
+        var attemptCount = 0
+        let baseURL = try XCTUnwrap(URL(string: "http://example.com"), "Unwrap fail")
+        do {
+            let result = try await sut.intercept(
+                HTTPRequest(
+                    method: .get,
+                    scheme: "http",
+                    authority: "example.com",
+                    path: "/test"
+                ),
+                body: nil,
+                baseURL: baseURL,
+                operationID: "testOperation"
+            ) { _, _, _ in
+                attemptCount += 1
+                throw URLError(URLError.Code(rawValue: notFound))
```

Comment:
???

Reply:
added enum RetryingMiddlewareError and changed throw error to RetryingMiddlewareError.error.